### PR TITLE
Defining "IN" and "NOT IN"  APIs in query module.Fixes apache/skywalking [BanyanDB]#9759

### PR DESCRIPTION
**Pull request by DevPJ9**
**Description**
Added "In" and "Not_In" parsers in "index_filter.go" so that the query module can implement them.

### CODE REVIEW

**Comment by Superskyyy** [comment]
It's better to make the PR title more informative than just an issue reference. Try describe what was changed in the actual context. 

Then you refer to the issue that you resolve in text body - Closes: #9759. It will automatically close for you once merged.

File: pkg/query/logical/index_filter.go [hunk 1]
```diff
@@ -146,6 +146,20 @@ func parseCondition(cond *model_v1.Condition, indexRule *database_v1.IndexRule,
 			and.append(newEq(indexRule, newBytesLiteral(b)))
 		}
 		return newNot(indexRule, and), []tsdb.Entity{entity}, nil
+	case model_v1.Condition_BINARY_OP_IN:
+		bb := expr.Bytes()
+		and := newOr(len(bb))
```
Comment (hanahmily) [inline pkg/query/logical/index_filter.go:151]: Renam the variable and other "and" to "or".

**Review by hanahmily — CHANGES_REQUESTED** [review]
Some nits

**Comment by hanahmily** [comment]
As @Superskyyy mentioned, you should tweak the title and refer to the issue to close it automatically.

I highly recommend that you should make all changes in this PR rather than splitting it into several tiny ones. This strategy helps contributors track changes. That means you could keep pushing more changes Here. Transforming this PR to a "draft" makes more sense.

Once all is ready and checks and tests are passed, you could turn this PR to the reviewable phase.

**Comment by DevPJ9** [comment]
@Superskyyy I have changed the title and description. Please reveiw.

**Pull request closed without merging**